packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -253,6 +254,35 @@ export class ResourceService {
       throw new Error(INVALID_RESOURCE_ID);
     }
 
+    const resourceWithProject = await this.prisma.resource.findUnique({
+      where: {
+        id: resource.id,
+      },
+      include: {
+        project: true,
+      },
+    });
```

Comment:
These 2 calls to the DB are redundant. 
You can already use the `resource.projectId` and `user.workspace.id` where these two data points are needed.